svgnative/ports/cairo/CairoSVGRenderer.cpp

```diff
@@ -0,0 +1,437 @@
+            cairo_set_source(mCairo, pat);
+        } else {
+            const auto& color = boost::get<Color>(fillStyle.paint);
+            cairo_set_source_rgba(mCairo,
+                                  color[0],
+                                  color[1],
+                                  color[2],
+                                  color[3] * fillStyle.fillOpacity * graphicStyle.opacity );
+        }
+
+        // Skia backend does not handle fillStyle.Rule yet
+        switch (fillStyle.fillRule) {
+        case WindingRule::kEvenOdd:
+            cairo_set_fill_rule (mCairo, CAIRO_FILL_RULE_EVEN_ODD);
+            break;
+        case WindingRule::kNonZero:
+        default:
+            cairo_set_fill_rule (mCairo, CAIRO_FILL_RULE_WINDING);
+        }
+
+        cairo_new_path (mCairo);
+        appendCairoSvgPath( mCairo, path );
+        cairo_fill (mCairo);
+    }
+    if (strokeStyle.hasStroke)
+    {
+        const auto& color = boost::get<Color>(strokeStyle.paint);
+        cairo_set_source_rgba(mCairo,
+                              color[0],
+                              color[1],
+                              color[2],
+                              color[3] * strokeStyle.strokeOpacity * graphicStyle.opacity );
+
+        cairo_set_line_width(mCairo, strokeStyle.lineWidth);
+
+        switch (strokeStyle.lineCap) {
+        case LineCap::kRound:
+            cairo_set_line_cap(mCairo, CAIRO_LINE_CAP_ROUND);
+            break;
+        case LineCap::kSquare:
+            cairo_set_line_cap(mCairo, CAIRO_LINE_CAP_SQUARE);
+            break;
+        case LineCap::kButt:
+        default:
+            cairo_set_line_cap(mCairo, CAIRO_LINE_CAP_BUTT);
+        }
+
+        switch (strokeStyle.lineJoin) {
+        case LineJoin::kRound:
+            cairo_set_line_join(mCairo, CAIRO_LINE_JOIN_ROUND);
+            break;
+        case LineJoin::kBevel:
+            cairo_set_line_join(mCairo, CAIRO_LINE_JOIN_BEVEL);
+            break;
+        case LineJoin::kMiter:
+        default:
+            cairo_set_line_join(mCairo, CAIRO_LINE_JOIN_MITER);
+        }
+
+        if (!strokeStyle.dashArray.empty()) {
+            std::vector<double> dashes;
+            for (auto dash : strokeStyle.dashArray)
+                 dashes.push_back(dash);
+            cairo_set_dash(mCairo, dashes.data(), strokeStyle.dashArray.size(), strokeStyle.dashOffset );
+        }
+
+        cairo_new_path (mCairo);
+        appendCairoSvgPath( mCairo, path );
+        cairo_stroke (mCairo);
+    }
+    Restore();
+}
+
+void CairoSVGRenderer::DrawImage(
+    const ImageData& image, const GraphicStyle& graphicStyle, const Rect& clipArea, const Rect& fillArea)
+{
+    SVG_ASSERT(mCairo);
+    Save(graphicStyle);
+    cairo_new_path (mCairo);
+    cairo_rectangle (mCairo, clipArea.x, clipArea.y, clipArea.width, clipArea.height);
+    cairo_clip(mCairo);
+
+    float w = image.Width();
```

Comment:
w and h are just used once. IMO no need for a temp variable.

Reply:
OK, I would remove them.